nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/StreamBulkProcessor.java

```diff
@@ -169,8 +202,16 @@ protected void processRecord(ComputationContext context, Record record) {
                         produceBucket(context, command.getAction(), bulkId, documentCount);
                     }
 
-                    kvStore.put(bulkId + STATE, RUNNING.toString());
-                    kvStore.put(bulkId + SCROLLED_DOCUMENT_COUNT, documentCount);
+                    Long scrollEndTime = Instant.now().toEpochMilli();
+
+                    BulkUpdate updates = new BulkUpdate();
+                    updates.put(bulkId + SCROLL_START_TIME, scrollStartTime.toString());
+                    updates.put(bulkId + SCROLL_END_TIME, scrollEndTime.toString());
+                    updates.put(bulkId + STATE, RUNNING.toString());
+                    updates.put(bulkId + SCROLLED_DOCUMENT_COUNT, String.valueOf(documentCount));
+                    Codec<BulkUpdate> updateCodec = Framework.getService(CodecService.class).getCodec(AVRO_CODEC,
+                            BulkUpdate.class);
+                    context.produceRecord(KVWRITER_STREAM_NAME, bulkId, updateCodec.encode(updates));
```

Comment:
it will have been easier to refactor the status (so it does not hold the command) and append the status so you have only 2 keys for each command: the "$comamndId":command, "$commandId:status": status